### ISSUE TITLE
Adding SPNEGO authentication support for Jetty Client

### ISSUE THREAD

**Issue by CaoManhDat** [eclipse/jetty.project#2868]
Although this kind of authentication seems not follow the HTTP rules but in production Spnego/Kerberos is widely used, i.e: Hadoop, Ambari. This is also a requirement feature for us when trying to move from Apache HttpClient to Jetty Client.

**Comment by sbordet** [comment]
See also #1708.

**Comment by sbordet** [comment]
Work is in progress for this issue, and we also took the chance to rewrite the server side, which allows us to test the whole scenario. We have working code that allows us to derive few conclusions.

SPNEGO requires a 401+200 for _every_ request.

This is a performance hit for `GET` requests, and a total bummer for `POST` requests, for obvious reasons.

Why people insist using SPNEGO with HTTP when it's so broken?

**Comment by sbordet** [comment]
The only way around is to have the server-side emit a non-guessable cookie after the initial SPNEGO handshake.

The server would then need to check this cookie and allow the request if the cookie matches.
This in turn requires the `HttpSession` in order to store the cookie value.
I guess we can use the last SPNEGO token sent by the server as the cookie value, so we don't need to generate it - although we still need to save it in the `HttpSession`.

@gregw should we implement this workaround or just leave SPNEGO as is (i.e. 401+200 for every request - with problems for `POST` requests with non-reproducible content)?

**Comment by sbordet** [comment]
See also: https://stackoverflow.com/questions/43456734/spnego-subsequent-calls-after-a-successful-negotiation-and-authentication

**Comment by CaoManhDat** [comment]
IIUC in Hadoop, after the authorization process, the server will then return a cookie to the client, for subsequence requests, the cookie will be used instead.

**Comment by sbordet** [comment]
@CaoManhDat I pushed working code to branch `jetty-9.4.x-2868-spnego_client`.
Can you please try it out?

@joshelser I have taken into account your observations in #1708 (use multiple Oids on server) in the new server-side implementation. Can you try it out?

To build:

```
$ git clone https://github.com/eclipse/jetty.project.git
$ cd jetty.project
$ git checkout jetty-9.4.x-2868-spnego_client
$ mvn clean install -DskipTests
```

Thanks!

**Comment by CaoManhDat** [comment]
Hi @sbordet I'm meeting some trouble when trying to integrate your implementation to Solr, since the configuration is provided through JAAS, i.e
```
Server {
 com.sun.security.auth.module.Krb5LoginModule required
  useKeyTab=true
  keyTab="/keytabs/zkhost1.keytab"
  storeKey=true
  doNotPrompt=true
  useTicketCache=false
  debug=true
  principal="zookeeper/host1@EXAMPLE.COM";
};
```
In your implementation it seems that we must provide all the parameters manually.

**Comment by sbordet** [comment]
@CaoManhDat you only have to provide the service `principal` and `keyTab` - all the other parameters are fixed and cannot be changed (otherwise it won't work).

And you have to provide the JAAS config name `Server`.

So with old code you need to A) set the system property with the JAAS config file path, and B) set the JAAS config name.

With the new code you need to A) set the service name, and B) set the keyTab path.

**Comment by CaoManhDat** [comment]
@sbordet Any reason for changing how the properties are set? Since in the old way it is quite simple and [official](https://docs.oracle.com/javase/7/docs/technotes/guides/security/jgss/tutorials/LoginConfigFile.html), just by setting: `-Djava.security.auth.login.config=jaas.conf`
The configuration file will then be read automatically
<img width="617" alt="screen shot 2018-09-18 at 5 55 06 pm" src="https://user-images.githubusercontent.com/258872/45682931-1bad3b80-bb6c-11e8-86b8-2564e6b4bc49.png">

**Comment by sbordet** [comment]
@CaoManhDat the old way was as official as the new way and it's not enough to only set the system property, unless you have hardcoded in the code the name of the JAAS configuration, `Server` (which by the way is not the default name; the default name is `com.sun.security.jgss.accept`).
And you need to ship an external file outside the web application (`jaas.conf`).
And you need to set a system property that precludes running 2 different servers/clients in the same JVM.
And you need to carefully write a number of properties in the JAAS configuration file that you must not change, so you don't know what can be changed and what must not be changed.

All in all, the old way is quite more complex than the new way.

Looking at the stack trace you posted, it is the client-side of the GSS negotiation. So the JAAS configuration should be called `Client`, not `Server`, and you want a principal that is a user name, not the service name (perhaps you showed the wrong `jaas.conf`? See how's confusing to use external files?).

My point being that on the client side you had to use system properties and external JAAS config files only because the implementation you were using was forcing you to do that.

Had that implementation done the opposite, you will be here asking me why you cannot set the principal via a setter :wink:

I don't know... seems all more complicated to me.
I can split the client-side classes into an abstract `SPNEGOAuthentication` with 2 subclasses, one for configuring with setters and one with nothing that relies on magic system properties.

I'm a bit confused if you want this on the client-side or on the server-side.

**Comment by CaoManhDat** [comment]
Sorry, the example `conf` file above is for the server-side. Here is an example for client
```
SolrClient {
 com.sun.security.auth.module.Krb5LoginModule required
 useKeyTab=true
 keyTab="/private/var/folders/9x/scy81vj16m12zn89zm3wdx6c0000gp/T/solr.cloud.TestSolrCloudWithKerberosAlt_FAC9AE7BA3C6EC54-001/tempDir-001/minikdc/keytabs"
 storeKey=true
 useTicketCache=false
 doNotPrompt=true
 debug=true
 principal="solr";
};
```

**Comment by CaoManhDat** [comment]
@sbordet You're right, this seems reasonable to me.

Solr in that case must include some way to automatically convert the old **magic** properties to the new one. Since we want users to have a pleasure experience when update from the old version of Solr to a new one.

**Comment by CaoManhDat** [comment]
@sbordet Thanks a lot, tests seem happy now, We will try to setup the authentication in a real cluster to check whether any errors can arise.

**Comment by sbordet** [comment]
@CaoManhDat as I said we can have a subclass that works like your old way, but we need more information, in particular: how was the JAAS config name `SolrClient` set on the implementation?

**Comment by CaoManhDat** [comment]
@sbordet `SolrClient` is set by a system property named `solr.kerberos.jaas.appname`

**Comment by sbordet** [comment]
Uhm, hardcoded system properties everywhere.

I'll think about this, but I prefer to not hardcode anything in Jetty.

If you can pass the JAAS config name to a subclass, then surely you can pass other configuration parameters so that there is no need to hardcode anything, right?

**Comment by CaoManhDat** [comment]
Right, I can pass other configuration parameters (with some casting and null checks).

**Comment by sbordet** [comment]
Ugh, no. I'll create a subclass so that you can do:

```java
new JAASSPNEGOAUthentication(clientAppName);
```

so that you only set the JAAS system property and you're good.

**Comment by CaoManhDat** [comment]
BTW: @gerlowskija helped on testing the integration in an AWS cluster with Kerberos setup. The result seems good. We are unable to find any problem. The current implementation seems solid!

**Comment by sbordet** [comment]
@CaoManhDat let me recap: 

1. you have tested the client-side implementation only, with `HttpClient`.
2. you are configuring the new Jetty `SPNEGOAuthentication` class using setters like the snippet you posted a couple of comments ago.
3. with 1. and 2. everything works fine for you.

What do you use on the server? The old Jetty implementation? Something else?
Would you move to the new Jetty implementation?

**Comment by sbordet** [comment]
@CaoManhDat there is another thing that needs to be configured on the client: the service name.
From your files above this is something like `zookeeper/host1@EXAMPLE.COM`.
How do you configure the service name on the client? Another system property?

**Comment by CaoManhDat** [comment]
@sbordet SolrCloud have 2 components
- Nodes which act like servers, provide REST API. When needed nodes will act like clients to talk to other nodes.
- Zookeeper quorum which is centralize configuration storage. Zookeeper nodes in this case will be server and Solr nodes will act like client.

All kind of servers/clients above are guarded by Kerberos. So the setup which contains `zookeeper/host1@EXAMPLE.COM` is used for config Zookeeper node rather than Solr nodes. That is another system properties (https://lucene.apache.org/solr/guide/6_6/kerberos-authentication-plugin.html#KerberosAuthenticationPlugin-SolrStartupParameters)
@gerlowskija : may be you will add more informations here?

Our current implementation of Kerberos on server side based on org.apache.hadoop.security.authentication.server.AuthenticationFilter
That seem works pretty well so far. So we don't want to change that in near future.

**Comment by sbordet** [comment]
@CaoManhDat eventually we decided to not implement the version with hardcoded system properties.
You showed that it was possible for you to use the configurable version, so we stuck with that.

This is now merged in the main line of Jetty, and the next release should be in October - targeting a couple of weeks.

Thanks!

**Comment by richardydhan** [comment]
Hi @sbordet & friends:

How to use the class SPNEGOAuthentication? any example? (we use jetty-client 11.0.20)
Appreciate it very much if any advice provided, we have to make this work soon.

Tried to use it for our proxy auth, but got "AuthenticationProtocolHandler : **Bad credentials for TunnelRequest[**CONNECT xxx.sandbox.my.salesforce.com:443 HTTP/1.1]@bb56179"

We use EMP (enterprise messaging platform) implemented with Jetty client/httpclient for an event listener to Salesforce platform. Added below class for auth:

----
        String uriString = "http://" + this.proxyHost + ":" +  this.proxyPort
        String cachePath = System.getenv("KRB5CCNAME")

        URI proxyUri = new URI(uriString)
        SPNEGOAuthentication auth = new SPNEGOAuthentication(proxyUri);
        auth.setUseTicketCache(true)
        auth.setTicketCachePath(Path.of(cachePath, ""))
        auth.setUserName("idi_kfk_t") //needed?
        auth.setServiceName("HTTP") // what's the service name? needed?
----
In EmpConnector constructor:
         AuthenticationStore authStore = httpClient.getAuthenticationStore();
        authStore.addAuthentication(parameters.getKerberosAuthentication());
----
2024-04-09T16:42:07.802Z DEBUG 1 --- [ent@21046afa-37] org.eclipse.jetty.http.HttpParser        : HEADER --> CONTENT
2024-04-09T16:42:07.802Z DEBUG 1 --- [ent@21046afa-37] org.eclipse.jetty.client.HttpReceiver    : Response headers HttpResponse[HTTP/1.1 407 Proxy Authentication Required]@4b048086
Proxy-Authenticate: NEGOTIATE
Proxy-Authenticate: NTLM
Cache-Control: no-cache
Pragma: no-cache
X-XSS-Protection: 1
Content-Type: text/html; charset=UTF-8
Proxy-Connection: close
Connection: close
Content-Length: 513
2024-04-09T16:42:07.803Z DEBUG 1 --- [ent@21046afa-37] org.eclipse.jetty.client.HttpReceiver    : Response demand=1/1, resume=false
2024-04-09T16:42:07.803Z DEBUG 1 --- [ent@21046afa-37] org.eclipse.jetty.client.HttpReceiver    : Response headers hasDemand=true HttpResponse[HTTP/1.1 407 Proxy Authentication Required]@4b048086
2024-04-09T16:42:07.803Z DEBUG 1 --- [ent@21046afa-37] org.eclipse.jetty.client.HttpReceiver    : Response content HttpResponse[HTTP/1.1 407 Proxy Authentication Required]@4b048086
DirectByteBufferR@40a54e58[p=272,l=785,c=16384,r=513]={HTTP/1.1 ...513\r\n\r\n<<<<!DOCTYPE html> CONTENT_END
2024-04-09T16:42:07.805Z DEBUG 1 --- [ent@21046afa-37] org.eclipse.jetty.http.HttpParser        : CONTENT_END --> END
2024-04-09T16:42:07.805Z DEBUG 1 --- [ent@21046afa-37] org.eclipse.jetty.http.HttpParser        : reset HttpParser{s=END,513 of 513}
2024-04-09T16:42:07.805Z DEBUG 1 --- [ent@21046afa-37] org.eclipse.jetty.http.HttpParser        : END --> START
2024-04-09T16:42:07.805Z DEBUG 1 --- [ent@21046afa-37] org.eclipse.jetty.client.HttpReceiver    : Response success HttpResponse[HTTP/1.1 407 Proxy Authentication Required]@4b048086
2024-04-09T16:42:07.805Z DEBUG 1 --- [ent@21046afa-37] org.eclipse.jetty.client.HttpExchange    : Terminated response for HttpExchange@3a7428ad{req=TunnelRequest[CONNECT instamed--jpmc.sandbox.my.salesforce.com:443 HTTP/1.1]@bb56179[TERMINATED/null] res=HttpResponse[HTTP/1.1 407 Proxy Authentication Required]@4b048086[TERMINATED/null]}, result: Result[TunnelRequest[CONNECT instamed--jpmc.sandbox.my.salesforce.com:443 HTTP/1.1]@bb56179 > HttpResponse[HTTP/1.1 407 Proxy Authentication Required]@4b048086] null
2024-04-09T16:42:07.805Z DEBUG 1 --- [ent@21046afa-37] org.eclipse.jetty.client.HttpReceiver    : Response complete HttpResponse[HTTP/1.1 407 Proxy Authentication Required]@4b048086, result: Result[TunnelRequest[CONNECT instamed--jpmc.sandbox.my.salesforce.com:443 HTTP/1.1]@bb56179 > HttpResponse[HTTP/1.1 407 Proxy Authentication Required]@4b048086] null
2024-04-09T16:42:07.880Z DEBUG 1 --- [ent@21046afa-37] org.eclipse.jetty.client.HttpChannel     : HttpExchange@3a7428ad{req=TunnelRequest[CONNECT instamed--jpmc.sandbox.my.salesforce.com:443 HTTP/1.1]@bb56179[TERMINATED/null] res=HttpResponse[HTTP/1.1 407 Proxy Authentication Required]@4b048086[TERMINATED/null]} disassociated true from HttpChannelOverHTTP@6b9a9e36(exchange=null)[send=HttpSenderOverHTTP@24c51cab(req=QUEUED,failure=null)[HttpGenerator@3f6f79de{s=START}],recv=HttpReceiverOverHTTP@5000ce05(rsp=IDLE,failure=null)[HttpParser{s=START,0 of -1}]]
2024-04-09T16:42:07.881Z DEBUG 1 --- [ent@21046afa-37] o.e.j.client.http.HttpChannelOverHTTP    : Closing, reason: http/1.1 - HttpConnectionOverHTTP@444f9dbb::SocketChannelEndPoint@4a886b42[{l=/100.127.29.38:39628,r=server-proxy.jpmchase.net/169.122.150.45:11443,OPEN,fill=-,flush=-,to=81/30000}{io=0/0,kio=0,kro=1}]->[HttpConnectionOverHTTP@444f9dbb(l:/100.127.29.38:39628 <-> r:server-proxy.jpmchase.net/169.122.150.45:11443,closed=false)=>HttpChannelOverHTTP@6b9a9e36(exchange=null)[send=HttpSenderOverHTTP@24c51cab(req=QUEUED,failure=null)[HttpGenerator@3f6f79de{s=START}],recv=HttpReceiverOverHTTP@5000ce05(rsp=IDLE,failure=null)[HttpParser{s=START,0 of -1}]]]
2024-04-09T16:42:07.881Z DEBUG 1 --- [ent@21046afa-37] o.e.j.c.http.HttpConnectionOverHTTP      : Shutdown HttpConnectionOverHTTP@444f9dbb::SocketChannelEndPoint@4a886b42[{l=/100.127.29.38:39628,r=server-proxy.jpmchase.net/169.122.150.45:11443,OSHUT,fill=-,flush=-,to=82/30000}{io=0/0,kio=0,kro=1}]->[HttpConnectionOverHTTP@444f9dbb(l:/100.127.29.38:39628 <-> r:server-proxy.jpmchase.net/169.122.150.45:11443,closed=true)=>HttpChannelOverHTTP@6b9a9e36(exchange=null)[send=HttpSenderOverHTTP@24c51cab(req=QUEUED,failure=null)[HttpGenerator@3f6f79de{s=START}],recv=HttpReceiverOverHTTP@5000ce05(rsp=IDLE,failure=null)[HttpParser{s=START,0 of -1}]]]
2024-04-09T16:42:07.881Z DEBUG 1 --- [ent@21046afa-37] o.e.j.c.http.HttpConnectionOverHTTP      : Closed HttpConnectionOverHTTP@444f9dbb::SocketChannelEndPoint@4a886b42[{l=null,r=null,CLOSED,fill=-,flush=-,to=82/30000}{io=0/0,kio=-1,kro=-1}]->[HttpConnectionOverHTTP@444f9dbb(l:null <-> r:null,closed=true)=>HttpChannelOverHTTP@6b9a9e36(exchange=null)[send=HttpSenderOverHTTP@24c51cab(req=QUEUED,failure=null)[HttpGenerator@3f6f79de{s=START}],recv=HttpReceiverOverHTTP@5000ce05(rsp=IDLE,failure=null)[HttpParser{s=START,0 of -1}]]]
2024-04-09T16:42:07.882Z DEBUG 1 --- [ent@21046afa-37] org.eclipse.jetty.client.HttpReceiver    : Request/Response succeeded: Result[TunnelRequest[CONNECT instamed--jpmc.sandbox.my.salesforce.com:443 HTTP/1.1]@bb56179 > HttpResponse[HTTP/1.1 407 Proxy Authentication Required]@4b048086] null, notifying [org.eclipse.jetty.client.AuthenticationProtocolHandler$AfterAuthenticationListener@4eb7f6cb, org.eclipse.jetty.client.AuthenticationProtocolHandler$AuthenticationListener@60c69ab]
2024-04-09T16:42:07.882Z DEBUG 1 --- [ent@21046afa-37] o.e.j.c.AuthenticationProtocolHandler    : Bad credentials for TunnelRequest[CONNECT instamed--jpmc.sandbox.my.salesforce.com:443 HTTP/1.1]@bb56179
2024-04-09T16:42:07.882Z DEBUG 1 --- [ent@21046afa-37] o.eclipse.jetty.client.HttpConversation  : Exchanges in conversation 2, override=null, listeners=[org.eclipse.jetty.client.AuthenticationProtocolHandler$AfterAuthenticationListener@4eb7f6cb, org.eclipse.jetty.client.HttpProxy$CreateTunnelPromise$TunnelListener@559a2429]
2024-04-09T16:42:07.883Z DEBUG 1 --- [ent@21046afa-37] o.e.jetty.client.AbstractConnectionPool  : Connection creation failed MultiEntry@15d2ea6b{PENDING,usage=0,multiplex=0,pooled=null}

org.eclipse.jetty.client.HttpResponseException: Unexpected HttpContentResponse[HTTP/1.1 407 Proxy Authentication Required - 513 bytes] for TunnelRequest[CONNECT xxx.sandbox.my.salesforce.com:443 HTTP/1.1]@bb56179
	at org.eclipse.jetty.client.HttpProxy$CreateTunnelPromise$TunnelListener.onHeaders(HttpProxy.java:253)
	at org.eclipse.jetty.client.ResponseNotifier.notifyHeaders(ResponseNotifier.java:95)
	at org.eclipse.jetty.client.ResponseNotifier.notifyHeaders(ResponseNotifier.java:87)
	at org.eclipse.jetty.client.ResponseNotifier.forwardEvents(ResponseNotifier.java:249)
	at org.eclipse.jetty.client.ResponseNotifier.forwardSuccess(ResponseNotifier.java:223)
	at org.eclipse.jetty.client.ResponseNotifier.forwardSuccessComplete(ResponseNotifier.java:229)
	at org.eclipse.jetty.client.AuthenticationProtocolHandler$AuthenticationListener.forwardSuccessComplete(AuthenticationProtocolHandler.java:273)
	at org.eclipse.jetty.client.AuthenticationProtocolHandler$AuthenticationListener.onComplete(AuthenticationProtocolHandler.java:150)
	at org.eclipse.jetty.client.ResponseNotifier.notifyComplete(ResponseNotifier.java:213)
	at org.eclipse.jetty.client.ResponseNotifier.notifyComplete(ResponseNotifier.java:205)
	at org.eclipse.jetty.client.HttpReceiver.terminateResponse(HttpReceiver.java:492)
	at org.eclipse.jetty.client.HttpReceiver.terminateResponse(HttpReceiver.java:472)
	at org.eclipse.jetty.client.HttpReceiver.responseSuccess(HttpReceiver.java:435)
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.messageComplete(HttpReceiverOverHTTP.java:386)
	at org.eclipse.jetty.http.HttpParser.handleContentMessage(HttpParser.java:593)
	at org.eclipse.jetty.http.HttpParser.parseContent(HttpParser.java:1736)
	at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:1565)
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.parse(HttpReceiverOverHTTP.java:221)
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.process(HttpReceiverOverHTTP.java:160)
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.receive(HttpReceiverOverHTTP.java:91)
	at org.eclipse.jetty.client.http.HttpChannelOverHTTP.receive(HttpChannelOverHTTP.java:97)
	at org.eclipse.jetty.client.http.HttpConnectionOverHTTP.onFillable(HttpConnectionOverHTTP.java:207)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:314)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:100)
	at org.eclipse.jetty.io.SelectableChannelEndPoint$1.run(SelectableChannelEndPoint.java:53)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.runTask(AdaptiveExecutionStrategy.java:421)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.consumeTask(AdaptiveExecutionStrategy.java:390)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.tryProduce(AdaptiveExecutionStrategy.java:277)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.produce(AdaptiveExecutionStrategy.java:193)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:969)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.doRunJob(QueuedThreadPool.java:1194)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1149)
	at java.base/java.lang.Thread.run(Thread.java:842)

2024-04-09T16:42:07.886Z DEBUG 1 --- [ent@21046afa-37] org.eclipse.jetty.client.HttpExchange    : Failed HttpExchange@63c6b863{req=HttpRequest[POST /cometd/58.0/handshake HTTP/1.1]@79486f38[COMPLETED/org.eclipse.jetty.client.HttpResponseException: Unexpected HttpContentResponse[HTTP/1.1 407 Proxy Authentication Required - 513 bytes] for TunnelRequest[CONNECT xxx.sandbox.my.salesforce.com:443 HTTP/1.1]@bb56179] res=HttpResponse[null 0 null]@88a5b07[COMPLETED/org.eclipse.jetty.client.HttpResponseException: Unexpected HttpContentResponse[HTTP/1.1 407 Proxy Authentication Required - 513 bytes] for TunnelRequest[CONNECT xxx.sandbox.my.salesforce.com:443 HTTP/1.1]@bb56179]}: req=true/rsp=true {}

****
If we generate a negotiator token by ourself and add it to the header as proxy-authorization like below:

JettyHttpClientTransport httpTransport = new JettyHttpClientTransport(parameters.longPollingOptions(), httpClient) {
            @Override
            protected void customize(Request request) {
                request.header(AUTHORIZATION, bearerToken); //doe Salesforce api
                request.header(HttpHeader.PROXY_AUTHORIZATION.asString(), negotiatorHeader); //for proxy
            }
        };

then we got below error:
AuthenticationProtocolHandler    : No authentication available for TunnelRequest[CONNECT xxx.sandbox.my.salesforce.com:443 HTTP/1.1]@5ed6b864
org.eclipse.jetty.client.HttpResponseException: Unexpected HttpContentResponse[HTTP/1.1 407 Proxy Authentication Required - 513 bytes] for TunnelRequest[CONNECT xxx.sandbox.my.salesforce.com:443 HTTP/1.1]@5ed6b864
	at org.eclipse.jetty.client.HttpProxy$CreateTunnelPromise$TunnelListener.onHeaders(HttpProxy.java:253)

This indicating the proxy-auth header not passed over when create the tunnel.

**Comment by joakime** [comment]
Jetty 11 is now at End of Community Support

* https://github.com/jetty/jetty.project/issues/10485

Please try again with Jetty 12, and if you still have problems open a new issue against Jetty 12, don't revive this one.
